File: 226-invert-binary-tree/226-invert-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function invertTree(root: TreeNode | null): TreeNode | null {
    return swapSibling(root);

};

function swapSibling(node: TreeNode) {
    const leftChild = node?.left;
    const rightChild = node?.right;
    if(leftChild || rightChild) {
        // node.left = rightChild;
        // node.right = leftChild;
        node.left = swapSibling(rightChild)
        node.right = swapSibling(leftChild);
     }
    return node;
    
    
}

File: 226-invert-binary-tree/226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        
        def swapTree(node):
            if not node:
                return None
            left, right = swapTree(node.right), swapTree(node.left)
            node.left = left
            node.right = right
            return node
        return swapTree(root)
    
            
        